api/cometbft/abci/v4/types.pb.go

```diff
@@ -0,0 +1,7003 @@
+	}
+	handler := func(ctx context.Context, req interface{}) (interface{}, error) {
+		return srv.(ABCIServer).PrepareProposal(ctx, req.(*RequestPrepareProposal))
+	}
+	return interceptor(ctx, in, info, handler)
+}
+
+func _ABCI_ProcessProposal_Handler(srv interface{}, ctx context.Context, dec func(interface{}) error, interceptor grpc.UnaryServerInterceptor) (interface{}, error) {
+	in := new(RequestProcessProposal)
+	if err := dec(in); err != nil {
+		return nil, err
+	}
+	if interceptor == nil {
+		return srv.(ABCIServer).ProcessProposal(ctx, in)
+	}
+	info := &grpc.UnaryServerInfo{
+		Server:     srv,
+		FullMethod: "/cometbft.abci.v4.ABCI/ProcessProposal",
+	}
+	handler := func(ctx context.Context, req interface{}) (interface{}, error) {
+		return srv.(ABCIServer).ProcessProposal(ctx, req.(*RequestProcessProposal))
+	}
+	return interceptor(ctx, in, info, handler)
+}
+
+func _ABCI_ExtendVote_Handler(srv interface{}, ctx context.Context, dec func(interface{}) error, interceptor grpc.UnaryServerInterceptor) (interface{}, error) {
+	in := new(v3.RequestExtendVote)
+	if err := dec(in); err != nil {
+		return nil, err
+	}
+	if interceptor == nil {
+		return srv.(ABCIServer).ExtendVote(ctx, in)
+	}
+	info := &grpc.UnaryServerInfo{
+		Server:     srv,
+		FullMethod: "/cometbft.abci.v4.ABCI/ExtendVote",
+	}
+	handler := func(ctx context.Context, req interface{}) (interface{}, error) {
+		return srv.(ABCIServer).ExtendVote(ctx, req.(*v3.RequestExtendVote))
+	}
+	return interceptor(ctx, in, info, handler)
+}
+
+func _ABCI_VerifyVoteExtension_Handler(srv interface{}, ctx context.Context, dec func(interface{}) error, interceptor grpc.UnaryServerInterceptor) (interface{}, error) {
+	in := new(v3.RequestVerifyVoteExtension)
+	if err := dec(in); err != nil {
+		return nil, err
+	}
+	if interceptor == nil {
+		return srv.(ABCIServer).VerifyVoteExtension(ctx, in)
+	}
+	info := &grpc.UnaryServerInfo{
+		Server:     srv,
+		FullMethod: "/cometbft.abci.v4.ABCI/VerifyVoteExtension",
+	}
+	handler := func(ctx context.Context, req interface{}) (interface{}, error) {
+		return srv.(ABCIServer).VerifyVoteExtension(ctx, req.(*v3.RequestVerifyVoteExtension))
+	}
+	return interceptor(ctx, in, info, handler)
+}
+
+func _ABCI_FinalizeBlock_Handler(srv interface{}, ctx context.Context, dec func(interface{}) error, interceptor grpc.UnaryServerInterceptor) (interface{}, error) {
+	in := new(RequestFinalizeBlock)
+	if err := dec(in); err != nil {
+		return nil, err
+	}
+	if interceptor == nil {
+		return srv.(ABCIServer).FinalizeBlock(ctx, in)
+	}
+	info := &grpc.UnaryServerInfo{
+		Server:     srv,
+		FullMethod: "/cometbft.abci.v4.ABCI/FinalizeBlock",
+	}
+	handler := func(ctx context.Context, req interface{}) (interface{}, error) {
+		return srv.(ABCIServer).FinalizeBlock(ctx, req.(*RequestFinalizeBlock))
+	}
+	return interceptor(ctx, in, info, handler)
+}
+
+var _ABCI_serviceDesc = grpc.ServiceDesc{
+	ServiceName: "cometbft.abci.v4.ABCI",
```

Comment:
As can be seen here again, the new package introduces a new gRPC service path, so wire compatibility is not even necessary to maintain for the gRPC service.

On the other hand, maybe it's worth doing what @thanethomson did in #818 and put the service definition into a dedicated package namespace, so that it could be versioned separately? We don't really need the new service path as long as all request and response messages valid with the old service can still be compatibly represented.